src/dune_file_watcher/dune_file_watcher.ml

```diff
@@ -340,12 +340,14 @@ let fswatch_backend () =
     User_error.raise [ Pp.text "Please install fswatch to enable watch mode." ]
 
 let select_watcher_backend () =
-  if Sys.linux then (
-    assert (Ocaml_inotify.Inotify.supported_by_the_os ());
-    `Inotify_lib)
-  else if Fsevents.available () then `Fsevents
-  else if Sys.win32 then `Fswatch_win
-  else fswatch_backend ()
+  if Fsevents.available () then `Fsevents
```

Comment:
my understanding is that `fsevents` is only available on macOS. 

If we were to perform that check in the `match` expression below, however, we'd need an extra clause for `Other`, that's why there's an `if` expression. Is that right?

Reply:
There's an if expression because we need to use `Fsevents.available` somehow. It's possible that we'd disable on some versions on macos so it could be useful